src/pybind/mgr/cephadm/module.py

```diff
@@ -1174,6 +1174,26 @@ def remove_host_label(self, host, label):
         self.log.info('Removed label %s to host %s' % (label, host))
         return 'Removed label %s from host %s' % (label, host)
 
+    @trivial_completion
+    def host_ok_to_stop(self, hostname: str):
+        if hostname not in self.cache.get_hosts():
+            raise OrchestratorError(f'Cannot find host "{hostname}"')
+
+        daemons = self.cache.get_daemons()
+        daemon_map = defaultdict(lambda: [])
+        for dd in daemons:
+            if dd.hostname == hostname:
+                daemon_map[dd.daemon_type].append(dd.daemon_id)
+
+        for daemon_type,daemon_ids in daemon_map.items():
+            if not self.cephadm_services[daemon_type].ok_to_stop(daemon_ids):
+                raise orchestrator.OrchestratorError(
+                        f'It is NOT safe to stop host {hostname}')
```

Comment:
we should probably return the error string returned by `ceph * ok-to-stop` here.